src/test/java/io/logz/MockLogzioBulkListener.java

```diff
@@ -149,10 +153,63 @@ public boolean checkForLogExistence(String token, String type, String loggerName
                 if (!jsonObject.get("message").getAsString().equals(message)) {
                     found = false;
                 }
+
+                if (checkHostname) {
+                    try {
+                        String hostname = InetAddress.getLocalHost().getHostName();
+
+                        if (!jsonObject.get("hostname").getAsString().equals(hostname)) {
+                            found = false;
+                        }
+
+                    } catch (UnknownHostException e) {
+                        logger.error("Could not get hostname, conciser as failure!", e);
+                        found = false;
+                    }
+                }
+                else {
+
+                    // Fail if we have hostname but not checking it
+                    try {
+                        if (!jsonObject.get("hostname").getAsString().isEmpty()) {
+
+                            found = false;
+                        }
+                    } catch (NullPointerException e) {
+
+                        logger.debug("Did not find hostname, thats ok.");
+                    }
+                }
+
+                if (additionalFields != null) {
+
+                    Map<String,String> slittedAdditionalValues = Splitter.on(';').omitEmptyStrings().withKeyValueSeparator('=').split(additionalFields);
```

Comment:
This shouldn't be like this.
You should simply call it with "foo=bar" and then check the json has "foo" field with value "bar". You can't duplicate the code into the test - it's an anti pattern. Same goes for below with the env: Set an environment you invent "blabla1234" with value 12345. And check the field you specify has that value